src/DotNetWorker.Core/Hosting/ServiceCollectionExtensions.cs

```diff
@@ -56,6 +56,9 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorkerCore(this ISe
             services.AddSingleton<IConverterContextFactory, DefaultConverterContextFactory>();
             services.AddSingleton<IInputConversionFeatureProvider, DefaultInputConversionFeatureProvider>();
             services.AddSingleton<IInputConverterProvider, DefaultInputConverterProvider>();
+            
+            // Input binding cache
+            services.AddScoped<IBindingCache<ConversionResult>, DefaultBindingCache<ConversionResult>>();
```

Comment:
We create a new scope when accessing `InstanceServices` property of DefaultFunctionContext.